numpy/lib/_info.py

```diff
@@ -0,0 +1,384 @@
+"""
+Array API Inspection namespace
+
+This is the namespace for inspection functions as defined by the array API standard.
+See https://data-apis.org/array-api/latest/API_specification/inspection.html
+for more details.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from numpy import (
```

Comment:
Let's import dtypes from `from numpy._core import ...`, to avoid cyclic dependency. I would also move contents of this file to `numpy/lib/introspect.py`.